code/modules/ninja/energy_katana.dm

```diff
@@ -1,20 +1,76 @@
-/obj/item/weapon/katana/energy
+/obj/item/weapon/dash
+	name = "abstract dash weapon"
+	var/max_charges = 3
+	var/current_charges = 3
+	var/charge_rate = 30 //In deciseconds
+	var/dash_toggled = FALSE
+
+	var/start_effect_type = /obj/effect/temp_visual/dir_setting/ninja/phase/out
+	var/end_effect_type = /obj/effect/temp_visual/dir_setting/ninja/phase
+	var/beam_icon = "blur"
```

Comment:
Also please beam_icon_state as var name if it's not a file to keep it consistent.